INSTALL.md

```diff
@@ -39,23 +39,45 @@ That's it - you should now have a working copy ready for development!
 
 However, running code may at this point generate warnings related to sub-optimal performance and security. The reason for this is twofold: some features have not yet shipped as part of the official TensorFlow distribution, and some features rely on [custom ops](https://www.tensorflow.org/guide/extend/op) that first needs to be compiled. We address both below but stress that these steps can be skipped for initial experiments.
 
-# Custom Ops
+# Compiling Custom Ops
 
 Certains operations, such as secure randomness generation, rely on C++ extentions of TensorFlow known as [custom ops](https://www.tensorflow.org/guide/extend/op). These come precompiled with the [tf-encrypted pip package](https://pypi.org/project/tf-encrypted/) but need to be manually compiled when installing from source code as we did above.
 
-To do so you need the following system tools installed: curl, libtool, automake, and g++. These can typically be installed using your system's package manager (see below) and once done allows for building the custom ops using:
+Tensorflow custom ops must be built in a docker container to avoid ABI compatibility issues. First, you'll have to pull down the docker image:
 
 ```
-(venv) ./tf-encrypted/ $ make build
-``` 
+./tf-encrypted/ $ docker pull tensorflow/tensorflow:custom-op
+```
+
+Note: Docker commands may have to be run with sudo!
+
+Once the docker image has been retrieved we need to open up a shell to input the following commands.
+
+```
+./tf-encrypted/ $ docker run -it -v `pwd`:/opt/tf_encrypted \
```

Comment:
this is only on linux right? ie no need to go through this on macOS?

Reply:
Yep don't think this will work with macos. Will update when I add the macos instructions.